osu.Game.Tournament/TournamentGame.cs

```diff
@@ -90,12 +101,14 @@ private void load(FrameworkConfigManager frameworkConfig, GameHost host)
                     {
                         RelativeSizeAxes = Axes.Both,
                         Child = new TournamentSceneManager()
-                    }
+                    },
+                    topMostOverlayContent = new Container { RelativeSizeAxes = Axes.Both }
                 }, drawables =>
                 {
                     loadingSpinner.Hide();
                     loadingSpinner.Expire();
 
+                    topMostOverlayContent.Add(dialogOverlay);
```

Comment:
This is weird, why is `dialogOverlay` synchronously added on the load complete callback of the _async load_? Can you put the `dialogOverlay` as a child of `topMostOverlayContent` from the start?

Reply:
This is the part I was kinda winging, I'm not sure how I ended up with the code in this form; IIRC doing what felt intuitive to me did not seem to work. I'll incorporate your other suggested changes first and revisit this soon